ui/src/shared/apis/query.ts

```diff
@@ -74,16 +74,21 @@ const handleQueryFetchStatus = async (
   }
 }
 
-const replace = async (
+export const replace = async (
   query: string,
   source: Source,
   templates: Template[],
   resolution: number
 ): Promise<string> => {
   try {
     query = replaceTemplates(query, templates)
-    const durationMs = await duration(query, source)
-    return replaceInterval(query, Math.floor(resolution / 3), durationMs)
+
+    if (query.includes(TEMP_VAR_INTERVAL)) {
```

Comment:
a slight optimization might be to move this logic into `replaceInterval`, and that that function checks whether there's interval and otherwise returns the query

Reply:
word, tho seems like `duration` is something that a `replaceIntervalAsync` function might call. just thinking out loud

Reply:
I like that